File: brain-move/games/base_game.py
import time


class BaseGame:
    """Classe base per tutti i giochi Brain-Move."""

    def __init__(self, difficulty_engine, scoring_engine):
        self.difficulty = difficulty_engine
        self.scoring = scoring_engine
        self.name = "base_game"
        self.display_name = "Gioco Base"
        self.running = False
        self.state = "idle"
        self.result = {}
        self.feedback_message = ""
        self.start_time = None
        self.duration_seconds = 0

    def start(self):
        self.running = True
        self.state = "playing"
        self.start_time = time.time()
        self.feedback_message = f"Preparati per {self.display_name}"

    def stop(self):
        self.running = False
        self.state = "finished"
        if self.start_time:
            self.duration_seconds = time.time() - self.start_time

    def update(self, pose_data=None, dt=0):
        raise NotImplementedError

    def get_state(self):
        return {
            "name": self.name,
            "display_name": self.display_name,
            "state": self.state,
            "feedback": self.feedback_message,
            "duration": self.duration_seconds,
            "result": self.result,
        }

    def is_finished(self):
        return self.state == "finished"

    def get_result(self):
        return self.result

    def get_progress(self):
        return 0

File: brain-move-web/src/games/base-game.ts
import { Difficulty } from '../engine/difficulty'
import { Scoring } from '../engine/scoring'
import type { PoseData, GameState, GameResult } from '../types'

export abstract class BaseGame {
  difficulty: Difficulty
  scoring: Scoring
  name = "base_game"
  displayName = "Gioco Base"
  running = false
  state: GameState = "idle"
  result: GameResult = { completed: false, accuracy: 0 }
  feedbackMessage = ""
  startTime: number | null = null
  durationSeconds = 0

  constructor(difficultyEngine: Difficulty, scoringEngine: Scoring) {
    this.difficulty = difficultyEngine
    this.scoring = scoringEngine
  }

  start(): void {
    this.running = true
    this.state = "playing"
    this.startTime = Date.now()
    this.feedbackMessage = `Preparati per ${this.displayName}`
  }

  stop(): void {
    this.running = false
    this.state = "finished"
    if (this.startTime) this.durationSeconds = (Date.now() - this.startTime) / 1000
  }

  abstract update(poseData?: PoseData | null, dt?: number): void

  getState() {
    return {
      name: this.name,
      displayName: this.displayName,
      state: this.state,
      feedback: this.feedbackMessage,
      duration: this.durationSeconds,
      result: this.result,
    }
  }

  isFinished(): boolean { return this.state === "finished" }
  getResult(): GameResult { return this.result }
  getProgress(): number { return 0 }
}
